src/codegen/types/conversions.js

```diff
@@ -44,9 +44,10 @@ const ETHEREUM_VALUE_TO_ASSEMBLYSCRIPT = [
   ],
   [/^string\[([0-9]+)?\]$/, 'Array<string>', code => `${code}.toStringArray()`],
 
-  // Tuple
+  // Tuples
 
   ['tuple', 'EthereumTuple', code => `${code}.toTuple()`],
+  [/^tuple\[([0-9]+)?\]$/, 'Array<EthereumTuple>', (code, type) => `${code}.toTupleArray<${type}>()`],
```

Comment:
So far we've been keeping the type casts out of this (see `toTuple()` above) and added ` as ${type}` in the code generation. I think I prefer that slightly.

Reply:
After looking at https://github.com/graphprotocol/graph-ts/pull/73 I see why you did this. Let's keep it!